app/policies/course_policy.rb

```diff
@@ -1,6 +1,25 @@
 class CoursePolicy
   attr_reader :user, :course
 
+  class TrainingProviderCourses
+    attr_reader :user, :scope
+
+    def initialize(user, scope)
+      @user = user
+      @scope = scope
+    end
+
+    def resolve
+      if user.admin?
+        scope.all
+      else
+        scope.where(
+          accrediting_provider_code: user.providers.pluck(:provider_code),
```

Comment:
im not sure is this correct

a user has a provider (its only this scope that should apply)
a provider has many courses
a course maybe accredited by another provider

Reply:
![image](https://user-images.githubusercontent.com/470137/72263436-c3c3ff00-3610-11ea-8c8c-84d0afda55c6.png)

Reply:
That was from a discussion I had with @misaka on Friday. When I was doing it it seemed like we should make sure that no courses were returned that weren't associated, via the training provider, to the accredited body. That is was happens but only as a result of the implementation, it's not being enforced by the authorisation system. I can remove the scope if we agree it's not necessary. Regarding the line 17 where clause... There isn't any way of knowing in the policy which `accredited_body` the user is viewing from so it seemed 'good enough' ™️ to allow them to see courses that any of their providers are the accrediting body for. The relationship is `has_many` in the code rather than 'has a'.

Reply:
https://github.com/DFE-Digital/manage-courses-backend/blob/3c2e7a0fe1c92550ee62c4e0e6f9d4ebaad3d9b0/app/models/provider.rb#L82

Reply:
is that what you was looking for?

Reply:
Discussed ^ off-line. Yes, that is what we need but additionally `current` and by provider id.

Reply:
There may be an issue with line 17 in that it doesn't check that accredited body the user is associated with is current for this recruitment cycle. Open to suggestions on that.

Reply:
Ya, I don't know about this yet. Let's have a chat about it.